src/main/java/org/padaiyal/aayvalar/statistics/StatisticsUtility.java

```diff
@@ -0,0 +1,248 @@
+package org.padaiyal.aayvalar.statistics;
+
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.DoubleSummaryStatistics;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.padaiyal.utilities.I18nUtility;
+
+/**
+ * Statistics utility library for retrieving basic statistical metrics.
+ */
+public class StatisticsUtility {
+
+  static {
+    I18nUtility.addResourceBundle(
+        StatisticsUtility.class,
+        StatisticsUtility.class.getSimpleName(),
+        Locale.US
+    );
+  }
+
+  /**
+   * Empty private constructor as this utility class is not meant to be used as an instance.
+   */
+  private StatisticsUtility() {
+  }
+
+  /**
+   * Gets the average of the provided collection.
+   *
+   * @param inputCollection The collection to get the mean from.
+   * @param meanType        The type of mean to calculate.
+   * @return The mean of the values inside a collection.
+   */
+  public static double getMean(Collection<? extends Number> inputCollection, MeanType meanType) {
```

Comment:
Add a few words explaining the different types of means, or at least link them to their corresponding wikipedia page..

Reply:
done